plugin/trino-snowflake/src/test/java/io/trino/plugin/snowflake/TestSnowflakeConnectorTest.java

```diff
@@ -0,0 +1,466 @@
+        }
+        // Error: Failed to insert data: SQL compilation error: error line 1 at position 130
+        if (typeName.equals("time")
+                || typeName.equals("time(6)")
+                || typeName.equals("timestamp(6)")) {
+            return Optional.empty();
+        }
+        // Error: not equal
+        if (typeName.equals("char(3)")) {
+            return Optional.empty();
+        }
+        return Optional.of(dataMappingTestSetup);
+    }
+
+    @Override
+    protected boolean isColumnNameRejected(Exception exception, String columnName, boolean delimited)
+    {
+        return nullToEmpty(exception.getMessage()).matches(".*(Incorrect column name).*");
+    }
+
+    @Override
+    protected MaterializedResult getDescribeOrdersResult()
+    {
+        // Override this test because the type of row "shippriority" should be bigint rather than integer for snowflake case
+        return resultBuilder(getSession(), VARCHAR, VARCHAR, VARCHAR, VARCHAR)
+                .row("orderkey", "bigint", "", "")
+                .row("custkey", "bigint", "", "")
+                .row("orderstatus", "varchar(1)", "", "")
+                .row("totalprice", "double", "", "")
+                .row("orderdate", "date", "", "")
+                .row("orderpriority", "varchar(15)", "", "")
+                .row("clerk", "varchar(15)", "", "")
+                .row("shippriority", "bigint", "", "")
+                .row("comment", "varchar(79)", "", "")
+                .build();
+    }
+
+    @Test
+    @Override
+    public void testShowColumns()
+    {
+        assertThat(query("SHOW COLUMNS FROM orders")).matches(getDescribeOrdersResult());
+    }
+
+    @Test
+    public void testViews()
+    {
+        String tableName = "test_view_" + randomNameSuffix();
+        onRemoteDatabase().execute("CREATE OR REPLACE VIEW tpch." + tableName + " AS SELECT * FROM tpch.orders");
+        assertQuery("SELECT orderkey FROM " + tableName, "SELECT orderkey FROM orders");
+        onRemoteDatabase().execute("DROP VIEW IF EXISTS tpch." + tableName);
+    }
+
+    @Test
+    @Override
+    public void testShowCreateTable()
+    {
+        // Override this test because the type of row "shippriority" should be bigint rather than integer for snowflake case
+        assertThat(computeActual("SHOW CREATE TABLE orders").getOnlyValue())
+                .isEqualTo("CREATE TABLE snowflake.tpch.orders (\n" +
+                        "   orderkey bigint,\n" +
+                        "   custkey bigint,\n" +
+                        "   orderstatus varchar(1),\n" +
+                        "   totalprice double,\n" +
+                        "   orderdate date,\n" +
+                        "   orderpriority varchar(15),\n" +
+                        "   clerk varchar(15),\n" +
+                        "   shippriority bigint,\n" +
+                        "   comment varchar(79)\n" +
+                        ")\n" +
+                        "COMMENT ''");
```

Comment:
Why here has `COMMENT` field even we declare not support `SUPPORTS_CREATE_TABLE_WITH_TABLE_COMMENT` and `SUPPORTS_COMMENT_ON_TABLE`